File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        sums = {}
        res = []
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in sums:
                res = [sums[diff], i]
            sums[nums[i]] = i
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int[] visited = new int[nums.length];
        int visitedSize = 0;
        for (int i = 0; i < nums.length; i++) {
            for (int j = 0; j < visitedSize; j++) {
                if (nums[i] == visited[j]) {
                    return true;
                }
            }
            visited[i] = nums[i];
            visitedSize++;
        }
        return false;
    }
}